src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Float.cs

```diff
@@ -89,23 +83,34 @@ public static bool TryFormat(float value, Span<byte> destination, out int bytesW
                 default:
                     return FormattingHelpers.TryFormatThrowFormatException(out bytesWritten);
             }
+            
+            Span<char> formatText = stackalloc char[StandardFormat.FormatStringLength];
+            int formattedLength = format.Format(formatText);
+            formatText = formatText.Slice(0, formattedLength);
 
-            string formatString = format.ToString();
-            string utf16Text = value.ToString(formatString, CultureInfo.InvariantCulture);
-            int length = utf16Text.Length;
-            if (length > destination.Length)
+            // We first try to format into a stack-allocated buffer, and if it succeeds, we can avoid
+            // all allocation.  If that fails, we fall back to allocating strings.  If it proves impactful,
+            // that allocation (as well as roundtripping from byte to char and back to byte) could be avoided by
+            // calling into a refactored Number.FormatSingle/Double directly.
+
+            Span<char> stackBuffer = stackalloc char[128]; // large enough to handle the majority cases
+            ReadOnlySpan<char> utf16Text = value.TryFormat(stackBuffer, out formattedLength, formatText, CultureInfo.InvariantCulture) ?
+                stackBuffer.Slice(0, formattedLength) :
+                value.ToString(new string(formatText), CultureInfo.InvariantCulture).AsSpan();
+
+            if (utf16Text.Length > destination.Length)
             {
                 bytesWritten = 0;
                 return false;
             }
 
-            for (int i = 0; i < length; i++)
+            for (int i = 0; i < utf16Text.Length; i++)
```

Comment:
Do we not have an optimized string copy operation for ASCII only text? This is highly vectorizable.

Reply:
I don't think so, but if we did or if you added one, we could certainly change this to use it.

Reply:
I might log a tracking bug for this (it should be markable as `up for grabs`).